crates/red_knot_python_semantic/src/types.rs

```diff
@@ -1537,6 +1566,11 @@ impl<'db> Type<'db> {
                 // TODO: implement tuple methods
                 todo_type!().into()
             }
+            Type::AlwaysTruthy | Type::AlwaysFalsy => {
+                // TODO: This should correctly handle scenarios involving negated type conditions,
+                // such as `int & ~AlwaysFalsy`.
+                todo_type!().into()
```

Comment:
I think this TODO really falls under the existing TODO for `Type::Intersection` above. I think we can correctly just fall back to member access on `object` here. The only exception would be that for `__bool__` we _could_ synthesize and return a callable type that returns `Literal[True]` or `Literal[False]`, but the latter would need to be a TODO for now, since we don't have general Callable types yet (just function literal types.)
```suggestion
                // TODO return `Callable[[], Literal[True/False]]` for `__bool__` access
                KnownClass::Object.to_instance(db).member(db, name)
```